pkg/cloudprovider/cloudprovider.go

```diff
@@ -122,10 +122,20 @@ func (c *CloudProvider) Create(ctx context.Context, nodeClaim *karpv1.NodeClaim)
 	if len(instanceTypes) == 0 {
 		return nil, cloudprovider.NewInsufficientCapacityError(fmt.Errorf("all requested instance types were unavailable during launch"))
 	}
-	instance, err := c.instanceProvider.Create(ctx, nodeClass, nodeClaim, instanceTypes)
+	instancePromise, err := c.instanceProvider.BeginCreate(ctx, nodeClass, nodeClaim, instanceTypes)
 	if err != nil {
 		return nil, fmt.Errorf("creating instance, %w", err)
```

Comment:
It looks like core now has [`cloudprovider.CreateError`](https://github.com/kubernetes-sigs/karpenter/blob/31eb6b511ee63fd2d04f382c0d82eec5cb6131d9/pkg/cloudprovider/types.go#L416-L420), which has a bit better mapping for reporting on NodeClaim conditions. May not be directly related to this PR, but possibly a good followup, for anything we return from Create?

Reply:
Let's tackle this later.